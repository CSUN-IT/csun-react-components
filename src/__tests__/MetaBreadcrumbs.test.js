import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import shortid from 'shortid';
import MetaBreadcrumbs from '../../src/components/meta/Breadcrumbs';

configure({ adapter: new Adapter() });

describe('MetaBreadcrumbs', () => {
  it('Should be defined', () => {
    expect(MetaBreadcrumbs).toBeDefined();
  });

  it('Renders component correctly', () => {
    const breadcrumbs = [
      {
        title: 'Home',
        to: '',
        home: true,
      },
      {
        title: 'IT',
        to: 'it',
      },
      {
        title: 'Giving',
      },
    ];

    shortid.generate = jest.fn(() => 0);

    const tree = shallow(
      <MetaBreadcrumbs className={'test'} breadcrumbItems={breadcrumbs} typecColor={'red'} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
