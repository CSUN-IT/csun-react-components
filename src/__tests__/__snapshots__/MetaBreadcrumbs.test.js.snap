// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaBreadcrumbs Renders component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MetaBreadcrumbs
    breadcrumbItems={
      Array [
        Object {
          "home": true,
          "title": "Home",
          "to": "",
        },
        Object {
          "title": "IT",
          "to": "it",
        },
        Object {
          "title": "Giving",
        },
      ]
    }
    className="test"
    typeColor=""
    typeStyle=""
    typecColor="red"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="breadcrumb__item breadcrumb__item--home "
        >
          <Link
            replace={false}
            to=""
          >
            <i
              aria-hidden="true"
              className="fa fa-home"
            />
          </Link>
          Home
        </li>,
        <li
          className="breadcrumb__item  "
        >
          
          IT
        </li>,
        <li
          className="breadcrumb__item  "
        >
          
          Giving
        </li>,
      ],
      "className": "test breadcrumbs  ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to=""
            >
              <i
                aria-hidden="true"
                className="fa fa-home"
              />
            </Link>,
            "Home",
          ],
          "className": "breadcrumb__item breadcrumb__item--home ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <i
                aria-hidden="true"
                className="fa fa-home"
              />,
              "replace": false,
              "to": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "fa fa-home",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": [Function],
          },
          "Home",
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            "IT",
          ],
          "className": "breadcrumb__item  ",
        },
        "ref": null,
        "rendered": Array [
          "",
          "IT",
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            "Giving",
          ],
          "className": "breadcrumb__item  ",
        },
        "ref": null,
        "rendered": Array [
          "",
          "Giving",
        ],
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="breadcrumb__item breadcrumb__item--home "
          >
            <Link
              replace={false}
              to=""
            >
              <i
                aria-hidden="true"
                className="fa fa-home"
              />
            </Link>
            Home
          </li>,
          <li
            className="breadcrumb__item  "
          >
            
            IT
          </li>,
          <li
            className="breadcrumb__item  "
          >
            
            Giving
          </li>,
        ],
        "className": "test breadcrumbs  ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to=""
              >
                <i
                  aria-hidden="true"
                  className="fa fa-home"
                />
              </Link>,
              "Home",
            ],
            "className": "breadcrumb__item breadcrumb__item--home ",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <i
                  aria-hidden="true"
                  className="fa fa-home"
                />,
                "replace": false,
                "to": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "fa fa-home",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": [Function],
            },
            "Home",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              "IT",
            ],
            "className": "breadcrumb__item  ",
          },
          "ref": null,
          "rendered": Array [
            "",
            "IT",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              "Giving",
            ],
            "className": "breadcrumb__item  ",
          },
          "ref": null,
          "rendered": Array [
            "",
            "Giving",
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
